daemons/attrd/attrd_cib.c

```diff
@@ -423,8 +522,29 @@ attrd_write_attribute(attribute_t *a, bool ignore_delay)
             }
         }
 
-        /* Initialize the status update XML */
-        xml_top = create_xml_node(NULL, XML_CIB_TAG_STATUS);
+        // Initiate a transaction for all the peer value updates
+        rc = the_cib->cmds->init_transaction(the_cib);
+        if (rc != pcmk_ok) {
+            crm_err("Failed to write %s (id %s, set %s): Could not initiate "
+                    "CIB transaction",
+                    a->id, pcmk__s(a->uuid, "n/a"), pcmk__s(a->set_id, "n/a"));
+            return;
+        }
+
+        if (pcmk__str_eq(a->set_type, XML_TAG_ATTR_SETS,
+                         pcmk__str_null_matches)) {
+            set_type = XML_TAG_ATTR_SETS;
+
+        } else if (pcmk__str_eq(a->set_type, XML_TAG_UTILIZATION,
+                                pcmk__str_none)) {
+            set_type = XML_TAG_UTILIZATION;
+
+        } else {
+            crm_err("Failed to write %s (id %s, set %s): Unknown set type %s",
+                    a->id, pcmk__s(a->uuid, "n/a"), pcmk__s(a->set_id, "n/a"),
+                    a->set_type);
+            goto done;
```

Comment:
It occurs to me we shouldn't create attributes with invalid set types in the first place (in `attrd_create_attribute()`), so this would simplify. We could skip the check for private attributes, but all others should be rejected as soon as possible.

Reply:
You're proposing:
* In `attrd_create_attribute()`, if `PCMK__XA_ATTR_SET_TYPE` is invalid **and** `PCMK__XA_ATTR_IS_PRIVATE` is false (indicating we'll never write it to the CIB), return `NULL`
* Drop the set type validation from `attrd_write_attribute()`

Is that correct?

Reply:
Yes, though we have `a->is_private`